src/components/search.js

```diff
@@ -0,0 +1,86 @@
+import { useEffect, useState } from 'react';
+import { useParams } from 'react-router';
+import { getCovidDataByCountry } from '../redux/country/countrySlice';
+import Header from './nav';
+
+const DetailsPage = () => {
+  const { name } = useParams();
+  const [state, setState] = useState(null);
+  useEffect(() => {
+    const getDetails = async () => {
+      const countryDetails = await getCovidDataByCountry(name);
+      setState(countryDetails);
+    };
+    getDetails();
+  }, [name]);
```

Comment:
When accessing the details page for some countries, I am presented with the following error. Please double-check your work to ensure that no errors popup when using your application🪲

<img width="863" alt="image" src="https://github.com/Mikey1-nuel/React-Capstone-Project/assets/37116322/da69109a-dd44-4491-91bd-a0186d0ec2a4">